addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -674,7 +674,30 @@ def write(self, values):
         employee_id = values.get('employee_id', False)
         if values.get('state'):
             self._check_approval_update(values['state'])
-        result = super(HolidaysAllocation, self).write(values)
+
+        if 'number_of_days_display' in values or 'number_of_hours_display' in values:
+            previous_consumed_leaves = self.employee_id._get_consumed_leaves(leave_types=self.holiday_status_id)
+            result = super(HolidaysAllocation, self).write(values)
```

Comment:
I think this isn't necessary anymore
```suggestion
            result = super().write(values)
```